streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```
@@ -540,6 +606,11 @@ private void initializeChangelogs(final Set<ChangelogMetadata> newPartitionsToRe
                 committedOffsets.get(partition) : Long.MAX_VALUE;
 
             if (endOffset != null && committedOffset != null) {
+                if (changelogMetadata.restoreEndOffset != null)
+                    throw new IllegalStateException("End offset for " + partition +
```

req: this is not covered in unit test.

---

Most of the places where IllegalStateException is thrown it is a place where we assert this should never happen (i.e. it did not depend on other modules) unless there's a bug, in which case we should always shout out and fail fast. So I think we do not need to have a unit test for such cases.

Other places of IllegalState is when the caller (e.g. ProcessorStateManager) had a bug like trying to register the same store twice, and only those cases that depend on other module bugs would have a test case.